tests/unit/test_color_refinement_helper.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+
+from color_refiment_helper import *
+
+
+class TestCRHelper(unittest.TestCase):
+
+    def create_graph_helper(self, edges: List[List["Integer"]]):
+        g = Graph(False)
+        vertices = {}
+        for edge in edges:
+            head, tail = edge
+            if head not in vertices:
+                vertices[head] = Vertex(g, label=head)
+                g.add_vertex(vertices[head])
+            if tail not in vertices:
+                vertices[tail] = Vertex(g, label=tail)
+                g.add_vertex(vertices[tail])
+            g.add_edge(Edge(vertices[head], vertices[tail]))
+        return g
+
+    def test_group_by(self):
+        # group_by(List[int]) groups by number
+        a = [1,2,3,2,3]
+        expected = {1: [1], 2: [2,2], 3: [3,3]}
+        self.assertEqual(expected, group_by(a))
+
+        # group_by(List[Vertex], key=Vertex.degree)
+        # 0 -- 1 -- 2
+        #                7 -- 4
+        #              /  \
+        #             6 - 5
+        #             \   /
+        #               3
+        g = self.create_graph_helper([[0,1],[1,2],[4,7],[3,5],[3,6],[5,7],[6,7],[5,6]])
+        vertices = sorted(g.vertices, key = lambda vertex: vertex.label)
+        expected = {1: [vertices[0], vertices[2], vertices[4]],
+                    2: [vertices[1], vertices[3]],
+                    3: [vertices[5], vertices[6], vertices[7]]}
+        actual = group_by(g.vertices, lambda v: v.degree)
+
+        self.assertEqual(expected.keys(), actual.keys())
+        for key in expected.keys():
+            self.assertTrue(compare(expected[key], actual[key], lambda v: v.label))
+
+        # group_by(dict{List}, key = lambda x: len(x)) groups by length of the lists
+        degrees = {1: [vertices[0], vertices[2], vertices[4]],
+                    2: [vertices[1], vertices[3]],
+                    3: [vertices[5], vertices[6], vertices[7]]}
+        expected = {2: [2], 3: [1,3]}
+        self.assertEqual(expected, group_by(degrees, lambda k: len(degrees[k])))
+
+        # group by amount of neighbours with a given color
+        # [x]: node with color/degree 1
+        # [0] -- 1 -- [2]
+        #                7 -- [4]
+        #              /  \
+        #             6 - 5
+        #             \   /
+        #               3
+        degree_coloring = group_by(g.vertices, lambda v: v.degree)
```

Comment:
Waarom hier `g.vertices` en elders gewoon `vertices`?

Reply:
`vertices` is de gesorteerde versie, dat was makkelijker voor het maken van de `expected` maps.